lib/src/rules/lines_longer_than_80_chars.dart

```diff
@@ -0,0 +1,170 @@
+
+    final lineCount = lineInfo.lineCount;
+    for (int i = 0; i < lineCount; i++) {
+      final start = lineInfo.getOffsetOfLine(i);
+      final end = i == lineCount - 1
+          ? node.end
+          : lineInfo.getOffsetOfLineAfter(start) - 1;
+      final length = end - start;
+      if (length > 80 && !allowedLines.contains(i + 1)) {
+        rule.reporter.reportErrorForOffset(rule.lintCode, start, length);
+      }
+    }
+  }
+}
+
+class _AllowedLongLineVisitor extends RecursiveAstVisitor {
+  _AllowedLongLineVisitor(this.lineInfo);
+
+  final LineInfo lineInfo;
+  final allowedLines = <int>[];
+
+  @override
+  visitStringInterpolation(StringInterpolation node) {
+    if (node.isMultiline) {
+      _handleMultilines(node);
+    } else {
+      final value = node.elements.map((e) {
+        if (e is InterpolationString) return e.value;
+        if (e is InterpolationExpression) return ' ' * e.length;
+      }).join();
+      _handleSingleLine(node, value);
+    }
+  }
+
+  @override
+  visitSimpleStringLiteral(SimpleStringLiteral node) {
+    if (node.isMultiline)
+      _handleMultilines(node);
+    else
+      _handleSingleLine(node, node.value);
+  }
+
+  _handleMultilines(SingleStringLiteral node) {
+    final startLine = lineInfo.getLocation(node.offset).lineNumber;
+    final endLine = lineInfo.getLocation(node.end).lineNumber;
+    for (var i = startLine; i <= endLine; i++) {
+      allowedLines.add(i);
+    }
+  }
+
+  _handleSingleLine(AstNode node, String value) {
+    if (_looksLikeUriOrPath(value)) {
+      final line = lineInfo.getLocation(node.offset).lineNumber;
+      allowedLines.add(line);
+    }
+  }
+}
+
+class _AllowedCommentVisitor extends GeneralizingAstVisitor {
+  _AllowedCommentVisitor(this.lineInfo);
+
+  final LineInfo lineInfo;
+  final allowedLines = <int>[];
+
+  @override
+  visitNode(AstNode node) {
+    final comments = <CommentToken>[]
+      ..addAll(_getPrecedingComments(node.beginToken))
+      ..addAll(_getPrecedingComments(node.endToken));
+    for (final comment in comments) {
+      final content = comment.toString();
+      final lines = [];
+      if (content.startsWith('///')) {
+        lines.add(content.substring(3));
+      } else if (content.startsWith('//')) {
+        lines.add(content.substring(2));
+      } else if (content.startsWith('/*')) {
+        // remove last slash before finding slash
+        lines.addAll(content.substring(2, content.length - 2).split('\n'));
+      }
+      for (var i = 0; i < lines.length; i++) {
+        final value = lines[i];
+        if (_looksLikeUriOrPath(value)) {
+          final line = lineInfo.getLocation(comment.offset).lineNumber + i;
+          allowedLines.add(line);
+        }
+      }
+    }
+    node.visitChildren(this);
+  }
+
+  List<CommentToken> _getPrecedingComments(Token token) {
+    var comments = <CommentToken>[];
+    var comment = token.precedingComments;
+    while (comment is CommentToken) {
```

Comment:
Both `token.preceedingComments` and `CommentToken.next` alway returns either `null` or a `CommentToken`, so this test isn't necessary. You might also be able to avoid creating a new list.

Reply:
Ouch! :-)